#19 Chore: setup docker

Niyonkuru1 opened the pull request:
### What does this PR do?
Add the docker config file for creating images of the application. This will be done at the end of the project, at least three images will be created and communication will be established between them.
### Description of Task to be completed?
- To create an image of backend
- To configure and run database in a docker container
- Both docker images can communicate while up and running
### How should this be manually tested?
`Docker` should be `installed` in your pc and configured to create and run images

`Clone the repo`

`checkout` to branch `ch-setup-docker-19
`

pull the changes by `git pull`

After checking that docker app is already installed,  Run `docker-compose up -d` to create an image from the docker-compose file and create container.

Then go to the browser and you should see the message being rendered. `IT IS NO MORE NECESSARY TO START SERVER LOCALLY`, docker handles them in the container

### Any background context you want to provide?
N/A

### What are the relevant pivotal tracker stories?
[Trello Board](https://trello.com/c/6CnMHtYC/19-setup-docker)

Screenshots (if appropriate)

[WIP #19]

Niyonkuru1 committed:
chore(setup-docker): configuring and integrating docker into the project

    - Refactoring the package.json to remove the unused packages
    - Adding script in package to run babel
    - Starting docker daemon after installing docker and configure it
    - Adding Dockerfile in the root folder and insert different command to be followed when building an image
    - Adding docker-compose.yml to the project to allow running the container in the best way by one command
    - Adding dockerignore and include files which is not required in the docker container
    - Adding small lines of code to permit testing easily

[WIP #19]

thepatrickniyo reviewed (changes requested):
Rebase to remove the conflicts

Niyonkuru1 committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/phantom-be-elite into ch-setup-docker-19

Niyonkuru1 committed:
chore(setup-docker):  adding postgres image and its configuration into the project

    - Grabing the postgres image from docker hub
    - adding the db service in the docker-compose.yml
    - Adding postgres-queries file which includes all files related to database connection in nodejs
    - containers connected eachother and talking.

[FINISHES #19]

Niyonkuru1 commented:
> Rebase to remove the conflicts

Yes I have did them. thank you

mukunzidd reviewed (approved):
@Niyonkuru1 LGTM.
